index-cn.bs

```diff
@@ -0,0 +1,1116 @@
+
+## 什么是 \[[This]] ## {#what-is-this}
+
+有时候，你可能会看到 <dfn lt="double brackets notation">\[[Notation]]</dfn> 被用于像 "Let <var ignore>proto</var> be <var ignore>obj</var>.\[[Prototype]]." 根据它出现的上下文，这个符号在技术上可以表示几种不同的东西，但你可能要花费很长时间才能理解，其实这个符号指的是某些无法通过JavaScript代码观察到的内部属性。  
+
+准确地说，它可以表示三种不同的东西，我将用规范中的示例来说明这一点。不过，现在也可以暂且跳过。
+
+### Record 字段 ### {#double-brackets-field-of-record}
+
+ECMAScript规范使 <dfn>Record</dfn> 这个术语来指代具有一组固定键的键值映射——有点像c语言中的结构。 [=Record=] 的每个键值对称为一个<dfn for=Record>field</dfn>. 因为 [=Records=] 只能出现在规范中，而不能出现在实际的JavaScript代码中，所以使用  [=double brackets notation|[[Notation]]=] 去引用 [=fields=] of a [=Record=].
+
+<div class=example>
+
+值得注意的是，[=Property Descriptors=] 属性描述符也是基于 [=Records=] with [=fields=] \[[Value]], \[[Writable]], \[[Get]], \[[Set]], \[[Enumerable]], and \[[Configurable]]. The [$ECMAScript/IsDataDescriptor$] [=abstract operation=] 广泛使用这种表示法：
+
+<div algorithm="to call the abstract operation IsDataDescriptor">
+
+当使用 [=Property Descriptor=] 调用抽象操作IsDataDescriptor 时，需要执行以下步骤:  
+
+1.  If |Desc| is **undefined**, return **false**.
+1.  If both |Desc|.\[[Value]] and |Desc|.\[[Writable]] are absent, return **false**.
+1.  Return **true**.
+
+</div>
+
+</div>
+
+在下一节中可以找到 [=Records=] 的另一个具体例子, [[#completion-records-and-shorthands]].
+
+<p class=note>Further reading: [=§6.2.1 The List and Record Specification Types=]</p>
+
+### JavaScript 对象的内部槽  ### {#double-brackets-internal-slot-of-javascript-object}
+
+JavaScript 对象可能会有一些所谓的 [=internal slots=] 而规范会使用这些槽来保存数据。像 [=Record/field|Record fields=], these [=internal slots=] 样，这些内部槽不能用JavaScript观察到，但可能其中的一些会通过特定的工具（如谷歌Chrome的DevTools）暴露出来。因此，使用 [=double brackets notation|[[Notation]]=] 表述 [=internal slots=].
+
+[=internal slots=] 的细节将在 [[#javascript-objects]]中讨论。现在，不要太担心它们的用途，先请注意下面的示例。  
+
+<div class=example>
+
+大多数JavaScript对象都有 \[[Prototype]] [=internal slot=]这么一个内部槽，它引用了当前对象所继承的对象。这个内部槽的值通常也就是 {{MDN/Object.getPrototypeOf()}} 方法返回的值。在 [$ECMAScript/OrdinaryGetPrototypeOf$] [=abstract operation=], 这个内部槽的 [=internal slot=] 值会被访问:
+
+<div algorithm="to call the abstract operation OrdinaryGetPrototypeOf">
+
+当对象O调用抽象操作 OrdinaryGetPrototypeOf 时，将执行以下步骤:
+
+1.  Return |O|.\[[Prototype]].
+
+</div>
+
+</div>
+
+Note: 对象的 [=Internal slots=] 和 [=Record/field|Record fields=] 的内部槽在外观上是相同的，但无论它是一个对象还是一个 [=Record=]，都可以通过查看这个表示法的实例主体(出现在'.'之前的部分)来消除歧义。从上下文环境来看通常是很容易区别的。
+
+###  JavaScript对象的内部方法 ### {#double-brackets-internal-method-of-javascript-object}
+
+JavaScript对象也有所谓的 [=internal methods=]. 像 [=internal slots=], 这些 [=internal methods=] 不能通过JavaScript直接观察到。因此，使用 [=double brackets notation|[[Notation]]=] to describe [=internal methods=].
+
+[=internal methods=] 的细节将在 [[#javascript-objects]] 讨论. 现在，不要太担心它们的用途，但请注意下面的示例。  
+
+<div class=example>
+
+所有JavaScript函数都有一个 \[[Call]] 内部方法来运行该函数。这个 [$ECMAScript/Call$] [=abstract operation=] 有以下步骤:  
+
+> 3.  Return ? |F|.\[[Call]](|V|, |argumentsList|).
+
+|F| i其中F是一个JavaScript函数对象。在这种情况下，F的 \[[Call]] 内部方法通过用参数V和argumentsList调用自身。  
```

Comment:
What's "i" at the beginning of the sentence? Also please use the variable syntax |V| and |argumentsList| at the end.

Reply:
fixed the typo and  |V| and |argumentsList|